cypress/page/filling-form.page.ts

```diff
@@ -0,0 +1,133 @@
+import { Hobbies } from "./hobbies-enum.page";
+
+class FillingFormPage {
+    private nameSelector: string
+    private lastNameSelector: string
+    private emailSelector: String
+    private genderSelector: String
+    private dateOfBirthSelector: String
+    private mobileNumberSelector: String
+    private hobbiesSelector: String
+    private currentAddressSelector: String
+    private pageURL: string
+    private submit: string
+    private stateSelector: string
+    private citySelector: string
+
+    constructor(){
+        this.nameSelector = "#firstName"
+        this.lastNameSelector = "#lastName"
+        this.emailSelector = "#userEmail"
+        this.genderSelector = "#genterWrapper .col-md-9"
+        this.dateOfBirthSelector = "#dateOfBirthInput"
+        this.mobileNumberSelector = "#userNumber"
+        this.hobbiesSelector = "#hobbiesWrapper .col-md-9"
+        this.currentAddressSelector = "#currentAddress"
+        this.stateSelector = "#state svg"
+        this.citySelector = "#city svg"
+        this.pageURL = "https://demoqa.com/automation-practice-form"
+        this.submit = "form"
+    }
+
+    public visitPage(): void{
+        cy.visit(this.pageURL)
+    }
+
+    public fillForm(personalInformation: any): void{
+            /* Get the hobbies number value 
+            according to the hobbies passed in the parameters */
+            const hobbiesValues = (
+                Object.keys(Hobbies).filter((val) =>
+                    isNaN(Number(val))) as (keyof typeof Hobbies)[]).map((key, value) => {
+                    if(personalInformation['hobbies'].includes(key))
+                        return value + 1;
+                }).filter(Number);
+            cy.get(this.nameSelector).type(personalInformation['name'])
```

Comment:
you can access to the properties using dot notation instead of array notation. So it is better:
`personalInformation.name` than `personalInformation['name']`